src/Mappers/Parameters/TypeHandler.php

```diff
@@ -124,6 +126,31 @@ private function getDocBlockPropertyType(DocBlock $docBlock, ReflectionProperty
         $varTags = $docBlock->getTagsByName('var');
 
         if (! $varTags) {
+            // If we don't have any @var tags, was this property promoted, and if so, do we have an
+            // @param tag on the constructor docblock?  If so, use that for the type.
+            if ($refProperty->isPromoted()) {
+                $refConstructor = $refProperty->getDeclaringClass()->getConstructor();
+                if (! $refConstructor) {
+                    return null;
+                }
+
+                $docBlock = $this->cachedDocBlockFactory->getDocBlock($refConstructor);
+                $paramTags = $docBlock->getTagsByName('param');
+                foreach ($paramTags as $paramTag) {
+                    if (! method_exists($paramTag, 'getVariableName')) {
```

Comment:
```suggestion
                    if (! $paramTag instanceof ParamDescriptor) {
```

I propose to check for expected type, not a existed method.

This allows IDE/SA to know about type of returned value from called method and avoid to have dead code in case of removed method in phpdocumentor (if it would not be catched with tests).

https://docs.phpdoc.org/3.0/classes/phpDocumentor-Descriptor-Tag-ParamDescriptor.html

Reply:
Sorry, commented from mobile :)